discovery-provider/src/queries/get_track_library.py

```diff
@@ -208,20 +213,23 @@ def _get_track_library(args: GetTrackLibraryArgs, session):
         # This branch covers added_date, or any other sort method
         if filter_type == LibraryFilterType.favorite:
             base_query = base_query.order_by(
-                sort_fn(Save.created_at), desc(TracksTable.track_id)
+                sort_fn(Save.created_at),
+                desc(cast(ColumnElement, TracksTable.track_id)),
             )
         elif filter_type == LibraryFilterType.repost:
             base_query = base_query.order_by(
-                sort_fn(Repost.created_at), desc(TracksTable.track_id)
+                sort_fn(Repost.created_at),
+                desc(cast(ColumnElement, TracksTable.track_id)),
             )
         elif filter_type == LibraryFilterType.purchase:
             base_query = base_query.order_by(
-                sort_fn(USDCPurchase.created_at), desc(TracksTable.track_id)
+                sort_fn(USDCPurchase.created_at),
+                desc(cast(ColumnElement, TracksTable.track_id)),
```

Comment:
strange we have to cast these.. are they not `ColumnElement` on the models already?

Reply:
should we be doing `TracksTable.track_id.desc()` now maybe instead? Or `TracksTable.c.track_id`?

Reply:
Bleh, not great. It's because the Aliased above. I may be warranted to upgrade to sqlalchemy 2.x because I think they handle this better, but for now I moved the cast to the definition of `TrackWithAggregatesAlias` and added a comment